lib/foreman_providers/compute_resource.rb

```diff
@@ -0,0 +1,35 @@
+module ForemanProviders
+  module ComputeResource
+    def self.included(base)
+      base.class_eval do
+        after_create   :create_provider
+        before_destroy :destroy_provider
+
+        private
+
+        def create_provider
+          provider_klass = foreman_type_to_provider_type.constantize
+          provider_klass.create!(
+            :name     => name,
+            :hostname => URI(url).host,
+          ).tap do |ems|
+            ems.update_authentication(:default => {:userid => user, :password => password})
+            ems.default_endpoint.update_attributes(:verify_ssl => 0)
```

Comment:
AFAIR for ovirt, we have the public key as part of the form (compute_resource.public_key) maybe you can use that and not disable ssl verification.

this is where maybe breaking down the logic per provider (using a subclass) could come handy.

Reply:
Yeah I saw that in the CR record, I don't think verifying ssl certs is a big deal for a demo.